data/transactions/logic/eval.go

```diff
@@ -2119,6 +2159,49 @@ func opSwitch(cx *EvalContext) error {
 	return nil
 }
 
+func opMatch(cx *EvalContext) error {
+	n := int(cx.program[cx.pc+1])
+	// stack contains the n sized match list and the single match value
+	if n+1 > len(cx.stack) {
+		return fmt.Errorf("match expects %d stack args while stack only contains %d", n+1, len(cx.stack))
+	}
+
+	top := len(cx.stack) - 1
+	matchVal := cx.stack[top]
+	cx.stack = cx.stack[:top]
+
+	top = len(cx.stack) - n
```

Comment:
Weird to call this `top`.  It's the base of the match values, right?

Reply:
Yes, I was thinking of `top` as representing the new top of the stack after the match cases are removed from the stack but I can see how that's imprecise/confusing.